src/kernels/kernel.base.ts

```diff
@@ -602,7 +602,12 @@ export abstract class BaseKernel implements IKernel {
             ]);
 
             // Have our debug cell script run first for safety
-            result.push(...debugCellScripts);
+            if (
+                isLocalConnection(this.kernelConnectionMetadata) &&
+                !this.configService.getSettings(undefined).useJupyterDebugger
```

Comment:
Is this the hidden setting? Forcing it to use ipykernel for debugging? Maybe the setting could be more descriptive? - like 'forceIPyKernelDebugger'

Reply:
Will fix in the next PR